source/main.cpp

```diff
@@ -94,10 +95,11 @@ using namespace Aws::Iot::DeviceClient::SensorPublish;
 
 const char *TAG = "Main.cpp";
 
-vector<Feature *> features;
+// vector<Feature *> features;
+FeatureRegistry features;
```

Comment:
Make it a `shared_ptr<FeatureRegistry>` and similar to `SharedCrtResourceManager` initialize in main.cpp.